M2/Macaulay2/packages/Permutations/Permutations.m2

```diff
@@ -0,0 +1,311 @@
+-- Sean Grate
+
+-----------------------------------------------------------------------------
+-- Local utilities
+-----------------------------------------------------------------------------
+
+to1Index = w -> (w / (i -> i+1))
+to0Index = w -> (w / (i -> i-1))
+
+-----------------------------------------------------------------------------
+-- Permutation type declarations and basic constructors
+-----------------------------------------------------------------------------
+
+Permutation = new Type of BasicList
+Permutation.synonym = "permutation"
+
+isValidPermutation = method(TypicalValue => Boolean)
+isValidPermutation List := Boolean => w -> (((sort w) == toList(1..#w)) and not (#w == 0))
+
+new Permutation from BasicList := (typeofPermutation,w) -> (
+    if not isValidPermutation(w) then error(toString w | " is not a valid permutation in one-line notation.");
+    w
+)
+
+permutation = method(TypicalValue => Permutation)
+permutation List := Permutation => w -> new Permutation from w
```

Comment:
Once you declared `TypicalValue => Permutation`, you don't need to list `Permutation => ` in front of every method.

(On the other hand, you should think about whether anyone else might want to use the method name `permutation` for a different purpose and have it return a different object, in which case fixing `TypicalValue => Permutation` would make their life harder)

Reply:
I didn't think about that side effect of `TypicalValue`, so it has been more carefully thought out and corrected in some new commits.